R/identifyPotentialDuplicates.R

```diff
@@ -0,0 +1,47 @@
+#' Identify Potential Duplicate Data Uploads
```

Comment:
Looks good. Is this a replacement for the original potential duplicates function, or in addition to? 

Here is some context/notes to consider for inclusion in the function description or vignette:

Notes: each unique result has a unique ResultIdentifier that should be included in all the result profiles. That should be all you need (for exact duplicates from the same submitting org), unless two different organizations submitted the same exact dataset to WQP (with only the org info being different). In addition to the org info, the site information (or really anything else org specific) could also be different if the two (or more) orgs are submitting this same dataset. In this scenario, the same datasets are being made available under different, but duplicate or nearby sites created separately by the submitting orgs. There are a lot of cases where the state and USGS (state) Water Science Center both are submitting the same datasets. This is something we may be able to try to address in the future through better communication between the states & USGS on who is submitting the data to WQP (but it might be more complicated than just a communication issue... it is possible some states feel they need (or maybe even have state requirements) to have the data linked to their orgs sites and their org names for use in their CWA assessments).

Reply:
I was having trouble finding an instance where the only thing different in a duplicate dataset was the organization metadata, but we can certainly keep both or even combine. This one is definitely still in development and needs tweaking. It should create a distance matrix between all sites with suspected duplicates and only flag those that are within X meters of one another with identical sample results for the same characteristic at the same date and time.